instrumentation/src/main/java/io/opentelemetry/android/internal/features/persistence/SimpleTemporaryFileProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.features.persistence;
+
+import io.opentelemetry.contrib.disk.buffering.internal.files.TemporaryFileProvider;
+import java.io.File;
+
+/**
+ * This class is internal and not for public use. Its APIs are unstable and can change at any time.
+ */
+public final class SimpleTemporaryFileProvider implements TemporaryFileProvider {
+    private final File tempDir;
+
+    public SimpleTemporaryFileProvider(File tempDir) {
+        this.tempDir = tempDir;
+    }
+
+    /** Creates a unique file instance using the provided prefix and the current time in millis. */
+    @Override
+    public File createTemporaryFile(String prefix) {
+        return new File(tempDir, prefix + "_" + System.currentTimeMillis() + ".tmp");
```

Comment:
What if the file name is a UUID? it could happen that 2 files would have the same name and it conflicts, likely hard to happen though.

Reply:
You can still sort files by timestamp if that's a concern, no need to rely on the file name.

Reply:
I think it's really difficult for that case to happen, although I also can't say it's impossible... So I've just made the change to use UUID just to be extra safe 👍 cheers.